saleor/app/manifest_validations.py

```diff
@@ -11,34 +12,56 @@
     get_permissions_enum_list,
     split_permission_codename,
 )
+from .const import (
+    AVAILABLE_APP_EXTENSION_CONFIGS,
+    EXTENSION_ENUM_MAP,
+    EXTENSION_FIELDS_MAP,
+)
 from .error_codes import AppErrorCode
-from .types import AppExtensionTarget, AppExtensionType, AppExtensionView
+from .types import AppExtensionOpenAs
 from .validators import AppURLValidator
 
-T_ERRORS = Dict[str, List[ValidationError]]
-
+logger = logging.getLogger(__name__)
 
-AVAILABLE_APP_EXTENSION_CONFIGS = {
-    AppExtensionType.DETAILS: {
-        AppExtensionView.PRODUCT: [
-            AppExtensionTarget.CREATE,
-            AppExtensionTarget.MORE_ACTIONS,
-        ]
-    },
-    AppExtensionType.OVERVIEW: {
-        AppExtensionView.PRODUCT: [
-            AppExtensionTarget.CREATE,
-            AppExtensionTarget.MORE_ACTIONS,
-        ]
-    },
-}
+T_ERRORS = Dict[str, List[ValidationError]]
 
 
 def _clean_app_url(url):
     url_validator = AppURLValidator()
     url_validator(url)
 
 
+def clean_extension_url(extension: dict, manifest_data: dict):
+    """Clean assigned extension url.
+
+    Make sure that format of url is correct based on the rest of manifest fields.
+    - url can start with '/' when one of these conditions is true:
+        a) extension.open_as == APP_PAGE
+        b) appUrl is provided
+    - url cannot start with protocol when openAs == "APP_PAGE"
+    """
+    extension_url = extension["url"]
+    open_as = extension.get("open_as") or AppExtensionOpenAs.POPUP
+    if extension_url.startswith("/"):
+        if open_as == AppExtensionOpenAs.APP_PAGE:
+            pass
```

Comment:
I think this statement is redundant. If we want to keep this anyway I would change it to `return`

Reply:
I agree. IMO we should change conditions in some way, or even move all these parts to a separate function.